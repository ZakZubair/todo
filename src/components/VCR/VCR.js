import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';
import DeleteIcon from '../../assets/img/delete.svg';

class VCR extends PureComponent {
  render() {
    const {
      recordingAvailable,
      isPlayingRecording,
      recordedTime,
      playRecordingTimeElapsed,
      isRecording,
      onClickCaptureRecording,
      onClickStopRecording,
      onClickPlayRecording,
      onClickClearRecording,
    } = this.props;

    let layout;
    let textAnimation = false;

    if (recordingAvailable && !isPlayingRecording) {
      layout = `Captured at ${recordedTime}`;
      textAnimation = true;
    } else if (recordingAvailable && isPlayingRecording) {
      layout = playRecordingTimeElapsed;
    } else if (isRecording) {
      layout = 'Recording...';
      textAnimation = false;
    } else {
      layout = 'Click to record';
    }

    return (
      <div className="vcr__sticky-header">
        <div className="vcr__sticky-header--display">
          <span className={textAnimation ? 'animated' : ''}>{layout}</span>
        </div>
        <div className="vcr__sticky-header--actions">
          {!isRecording && !recordingAvailable && (
            <Button className="button__record" onClick={onClickCaptureRecording} aria-label="Record" />
          )}
          {isRecording && (
            <Button className="button__stop" onClick={onClickStopRecording} aria-label="Stop Recording" />
          )}
          {recordingAvailable && (
            <Button className={`button__play ${isPlayingRecording ? 'button__play__playing' : ''}`} onClick={onClickPlayRecording} aria-label="Play Recording" />
          )}
          {recordingAvailable && !isPlayingRecording && (
            <Button className="button__clear" onClick={onClickClearRecording}>
              <img src={DeleteIcon} alt="delete" />
            </Button>
          )}
        </div>
      </div>
    );
  }
}

VCR.propTypes = {
  recordingAvailable: PropTypes.bool.isRequired,
  isPlayingRecording: PropTypes.bool.isRequired,
  recordedTime: PropTypes.string.isRequired,
  isRecording: PropTypes.bool.isRequired,
  onClickCaptureRecording: PropTypes.func.isRequired,
  onClickStopRecording: PropTypes.func.isRequired,
  onClickPlayRecording: PropTypes.func.isRequired,
  onClickClearRecording: PropTypes.func.isRequired,
  playRecordingTimeElapsed: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
};
export default VCR;
