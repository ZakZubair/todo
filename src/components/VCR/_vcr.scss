.vcr {
  &__sticky-header {
    display: flex;
    flex-direction: column;
    position: fixed;
    top: 0;
    right: 0;
    width: rem(192);
    height: rem(84);
    background-color: $white-80;
    border-radius: 0;
    box-shadow: 0 rem(30) rem(56) rem(6) rgba(0, 0, 0, 0.1);
    // Breakpoint for small screen
    @include breakpoint-sm {
      width: 100vw;
    }

    &--display {
      position: relative;
      padding: rem(4) 0 rem(4);
      text-align: center;
      width: 100vw;
      max-width: 100%;
      overflow-x: hidden;
      height: rem(20);
      margin-bottom: rem(4);
      background-color: $bright-gray;
      span {
        margin: 0;
        color: $turquoise;
        font-size: rem(14);

        &.animated {
          // text-align: right;
          position: absolute;
          white-space: nowrap;
          will-change: transform;
          animation: marquee 8s linear infinite;
          @include breakpoint-sm {
            animation: marquee 6s linear infinite;
          }
        }
      }
    }

    &--actions {
      display: flex;
      align-items: center;
      justify-content: center;
      padding-top: rem(4);
    }
  }
}
