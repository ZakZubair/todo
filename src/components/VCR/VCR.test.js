import React from 'react';
import renderer from 'react-test-renderer';
import VCR from './VCR';

const mockCallBack = jest.fn();

const commonProps = {
  onClickCaptureRecording: mockCallBack,
  onClickStopRecording: mockCallBack,
  onClickPlayRecording: mockCallBack,
  onClickClearRecording: mockCallBack,
};

describe('test VCR component', () => {
  it('render record button correctly', () => {
    const VCRComponent = renderer.create(
      <VCR
        recordingAvailable={false}
        isPlayingRecording={false}
        isRecording={false}
        recordedTime=""
        playRecordingTimeElapsed=""
        {...commonProps}
      />,
    ).toJSON();
    expect(VCRComponent).toMatchSnapshot();
  });

  it('render stop button correctly', () => {
    const VCRComponent = renderer.create(
      <VCR
        recordingAvailable={false}
        isPlayingRecording={false}
        isRecording
        recordedTime=""
        playRecordingTimeElapsed=""
        {...commonProps}
      />,
    ).toJSON();
    expect(VCRComponent).toMatchSnapshot();
  });

  it('render play button correctly', () => {
    const VCRComponent = renderer.create(
      <VCR
        recordingAvailable
        isPlayingRecording={false}
        isRecording={false}
        recordedTime=""
        playRecordingTimeElapsed=""
        {...commonProps}
      />,
    ).toJSON();
    expect(VCRComponent).toMatchSnapshot();
  });

  it('render playing correctly', () => {
    const VCRComponent = renderer.create(
      <VCR
        recordingAvailable
        isPlayingRecording
        isRecording
        recordedTime=""
        playRecordingTimeElapsed="0:10"
        {...commonProps}
      />,
    ).toJSON();
    expect(VCRComponent).toMatchSnapshot();
  });
});
