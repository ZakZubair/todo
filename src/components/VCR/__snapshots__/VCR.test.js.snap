// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test VCR component render play button correctly 1`] = `
<div
  className="vcr__sticky-header"
>
  <div
    className="vcr__sticky-header--display"
  >
    <span
      className="animated"
    >
      Captured at 
    </span>
  </div>
  <div
    className="vcr__sticky-header--actions"
  >
    <button
      className="button__play "
      onClick={[MockFunction]}
      type="button"
    />
    <button
      className="button__clear"
      onClick={[MockFunction]}
      type="button"
    >
      <img
        alt="delete"
        src={
          Object {
            "process": [Function],
          }
        }
      />
    </button>
  </div>
</div>
`;

exports[`test VCR component render playing correctly 1`] = `
<div
  className="vcr__sticky-header"
>
  <div
    className="vcr__sticky-header--display"
  >
    <span
      className=""
    >
      0:10
    </span>
  </div>
  <div
    className="vcr__sticky-header--actions"
  >
    <button
      className="button__stop"
      onClick={[MockFunction]}
      type="button"
    />
    <button
      className="button__play button__play__playing"
      onClick={[MockFunction]}
      type="button"
    />
  </div>
</div>
`;

exports[`test VCR component render record button correctly 1`] = `
<div
  className="vcr__sticky-header"
>
  <div
    className="vcr__sticky-header--display"
  >
    <span
      className=""
    >
      Click to record
    </span>
  </div>
  <div
    className="vcr__sticky-header--actions"
  >
    <button
      className="button__record"
      onClick={[MockFunction]}
      type="button"
    />
  </div>
</div>
`;

exports[`test VCR component render stop button correctly 1`] = `
<div
  className="vcr__sticky-header"
>
  <div
    className="vcr__sticky-header--display"
  >
    <span
      className=""
    >
      Recording...
    </span>
  </div>
  <div
    className="vcr__sticky-header--actions"
  >
    <button
      className="button__stop"
      onClick={[MockFunction]}
      type="button"
    />
  </div>
</div>
`;
